File: Data Structures & Algorithms/longest-increasing-subsequence/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    lengthOfLIS(nums) {
        let dp = Array(nums.length).fill(1)
        let ret = 0
        for(let i = nums.length - 1; i >= 0; i--){
            let res = 1;
            for(let j = i + 1; j < nums.length; j++){
                if(nums[i] < nums[j]){
                    // res = Math.max(res, Math.max(dp[i], dp[j]) + 1)
                    // // OR
                    dp[i] = Math.max(dp[i], dp[j] + 1)
                }
            }
            // dp[i] = res
            ret = Math.max(dp[i], ret)
        }   
        return ret
    }
}

File: Data Structures & Algorithms/house-robber/submission-2.java
class Solution {
    public int rob(int[] nums) {
        int rob1 = 0;
        int rob2 = 0;
        // [rob1, rob2, n, n+1, ...]
        for(int n: nums){
            int tmp = Math.max(rob1 + n, rob2);
            rob1 = rob2;
            rob2 = tmp;
        }
        return rob2;
    }
}
